src/Pages/ProductEdit.tsx

```diff
@@ -0,0 +1,274 @@
+import { object, string, number } from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Divider,
+  TextField,
+  Typography,
+  MenuItem,
+  styled,
+} from "@mui/material";
+import axios from "axios";
+import { useNavigate, useParams } from "react-router-dom";
+import { TerpenEnum } from "../types/Terpen";
+import { GeneticsEnum } from "../types/GeneticsEnum";
+import { useToken } from "../hooks/useToken";
+import { Spinner } from "../components/Spinner";
+
+const productSchema = object({
+  producentName: string().min(4).max(20).required("This field is required"),
+  strainName: string().min(4).max(20).required("This field is required"),
+  genetics: string().min(4).max(15).required(),
+  terpen: string().required(),
+  thcLevel: number().positive().integer().min(1).required(),
+  cbdLevel: number().positive().integer().min(1).required(),
+  description: string().required(),
+});
+
+type FormValues = {
+  producentName: string;
+  strainName: string;
+  genetics: string;
+  terpen: string;
+  thcLevel: number;
+  cbdLevel: number;
+  description: string;
+};
+
+const CustomCard = styled(Card)({
+  display: "flex",
+  justifyContent: "center",
+  alignItems: "center",
+  borderRadius: "16px",
+  padding: 25,
+  flexWrap: "wrap",
+});
+
+export const ProductEdit = () => {
+  const { id } = useParams(); // pobieram ID z URL
+  const navigate = useNavigate();
+
+  const [product, setProduct] = useState<FormValues | null>(null);
+  const [terpenOptions, setTerpenOptions] = useState<
+    { label: string; value: string }[]
+  >([]);
+  const [geneticsOptions, setGeneticsOptions] = useState<
+    { label: string; value: string }[]
+  >([]);
+
+  const form = useForm<FormValues>({
+    resolver: yupResolver(productSchema),
+    mode: "all",
+    defaultValues: {
+      producentName: "",
+      strainName: "",
+      genetics: GeneticsEnum.INDICA,
+      terpen: TerpenEnum.CARIOPHILEN,
+      thcLevel: 0,
+      cbdLevel: 0,
+      description: "",
+    },
+  });
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    formState: { errors },
+  } = form;
+
+  useEffect(() => {
+    // pobieram produktt po ID
+    const fetchProductData = () => {
+      axios
+        .get(`http://localhost:4000/product/${id}`)
+        .then((response) => {
+          const product = response.data;
+          setProduct(product);
+          // ustawiam w formie pobrane dane z backendu jako defaultowe
+          setValue("producentName", product.producentName); //Updatin the "description" field in the form with the description property from the fetched product data
+          setValue("strainName", product.strainName);
+          setValue("genetics", product.genetics);
+          setValue("terpen", product.terpen);
+          setValue("thcLevel", product.thcLevel);
+          setValue("cbdLevel", product.cbdLevel);
+          setValue("description", product.description);
```

Comment:
```suggestion
       const keys = Object.keys(product);
       keys.forEach((key) => setValue(key, product[key])
       )
```
jesli sa jakies klucze, ktorych nie ustawiasz to wczesniej:
`const {poleKtoregoNieUstawiam, ...rest} = product;`